src/jvmTest/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriterTest.kt

```diff
@@ -130,5 +133,121 @@ class CsvFileWriterTest : WordSpec() {
+                val row = listOf(
+                    LocalDate.of(2019, 8, 19),
+                    LocalDateTime.of(2020, 9, 20, 14, 32, 21)
+                )
+                val expected = "2019-08-19,2020-09-20T14:32:21\r\n"
+                csvWriter().openAsync(testFileName) {
+                    writeRow(row)
+                }
+                val actual = readTestFile()
+                actual shouldBe expected
+            }
+            "suspend write row from variable arguments" {
+
+                val date1 = LocalDate.of(2019, 8, 19)
+                val date2 = LocalDateTime.of(2020, 9, 20, 14, 32, 21)
+
+                val expected = "a,b,c\r\n"+
+                        "d,e,f\r\n"+
+                        "1,2,3\r\n"+
+                        "2019-08-19,2020-09-20T14:32:21\r\n"
+                csvWriter().openAsync(testFileName) {
+                    writeRow("a", "b", "c")
+                    writeRow("d", "e", "f")
+                    writeRow(1,2,3)
+                    writeRow(date1, date2)
+                }
+                val actual = readTestFile()
+                actual shouldBe expected
+            }
+            "suspend write all Sequence data" {
+                val rows = listOf(listOf("a", "b", "c"), listOf("d", "e", "f")).asSequence()
+                val expected = "a,b,c\r\nd,e,f\r\n"
+                csvWriter().openAsync(testFileName) {
+                    writeRows(rows)
+                }
+                val actual = readTestFile()
+                actual shouldBe expected
+            }
+        }
+        "suspeend writeAll method" should {
+            "suspend write all escaped field when a field contains quoteChar in it" {
+                val rows = listOf(listOf("a", "\"b", "c"), listOf("d", "e", "f\""))
+                val expected = "a,\"\"\"b\",c\r\nd,e,\"f\"\"\"\r\n"
+                csvWriter().openAsync(testFileName) {
+                    writeRows(rows)
+                }
+                val actual = readTestFile()
+                actual shouldBe expected
+            }
+            "suspend write all escaped field when a field contains delimiter in it" {
+                val rows = listOf(listOf("a", ",b", "c"), listOf("d", "e", "f,"))
+                val expected = "a,\",b\",c\r\nd,e,\"f,\"\r\n"
+                csvWriter().openAsync(testFileName) {
+                    writeRows(rows)
+                }
+                val actual = readTestFile()
+                actual shouldBe expected
+            }
+            "suspend write all quoted field when a field contains cr or lf in it" {
+                val rows = listOf(listOf("a", "\nb", "c"), listOf("d", "e", "f\r\n"))
+                val expected = "a,\"\nb\",c\r\nd,e,\"f\r\n\"\r\n"
+                csvWriter().openAsync(testFileName) {
+                    writeRows(rows)
+                }
+                val actual = readTestFile()
+                actual shouldBe expected
+            }
+        }
```

Comment:
Thanks for implementing it, but these tests are not for async behavior, but for CsvWriter logic.
So, it would be better to delete them.

Reply:
Resolved on [1e92c06](https://github.com/doyaaaaaken/kotlin-csv/pull/73/commits/1e92c06f8a09126d20f4f625f91f2683f426c47e)